pkgs/top-level/all-packages.nix

```diff
@@ -32265,6 +32269,15 @@ with pkgs;
 
   qrencode = callPackage ../development/libraries/qrencode {
     inherit (darwin) libobjc;
+    # Needed to break dependency cycles in systemd. SDL2 is only needed for
+    # tests in qrencode. This also reduces the number of rebuilds for systemd.
+    SDL2 = SDL2.override {
+      udevSupport = false;
+      dbusSupport = false;
+      pipewireSupport = false;
+      libdecorSupport = false;
+      pulseaudioSupport = false;
+    };
```

Comment:
Or simpler: we just copy config/no-x-libs and disable the check entirely and move it to passthru

Reply:
Good idea!